File: apps/board/web/src/types/memo.ts
export interface Memo {
  id: string;
  subject: string;
  body: string;
  tags: string[];
  contact_id: string | null;
  lead_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface MemoCreatePayload {
  subject: string;
  body: string;
  tags?: string[];
  contact_id?: string | null;
  lead_id?: string | null;
}

export type MemoUpdatePayload = Partial<MemoCreatePayload>;

File: apps/board/backend/app/models/memo.py
from __future__ import annotations

from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class MemoBase(BaseModel):
    subject: str
    body: str
    tags: List[str] = Field(default_factory=list)
    contact_id: Optional[UUID] = None
    lead_id: Optional[UUID] = None


class MemoCreate(MemoBase):
    pass


class MemoUpdate(BaseModel):
    subject: Optional[str] = None
    body: Optional[str] = None
    tags: Optional[List[str]] = None
    contact_id: Optional[UUID] = None
    lead_id: Optional[UUID] = None


class MemoRead(MemoBase):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    created_at: datetime
    updated_at: datetime
